src/main/java/codechicken/nei/ItemHistoryPanel.java

```diff
@@ -0,0 +1,206 @@
+package codechicken.nei;
+
+import static codechicken.lib.gui.GuiDraw.drawRect;
+
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.item.ItemStack;
+
+import org.lwjgl.opengl.GL11;
+
+import codechicken.nei.ItemPanel.ItemPanelSlot;
+import codechicken.nei.recipe.GuiCraftingRecipe;
+import codechicken.nei.recipe.GuiRecipe;
+import codechicken.nei.recipe.GuiUsageRecipe;
+import codechicken.nei.recipe.StackInfo;
+
+public class ItemHistoryPanel extends Widget {
+
+    protected int mouseDownSlot = -1;
+    protected ItemsGrid grid;
+
+    public ItemHistoryPanel() {
+        grid = new ItemsGrid();
+    }
+
+    public void draw(int mousex, int mousey) {
+
+        if (NEIClientConfig.getIntSetting("inventory.history.splittingMode") == 0) {
```

Comment:
Can you cache these values instead of getting them every render tick?

Reply:
many NEIClientConfig calculated every tick. we need a systematic solution to this problem

![image](https://github.com/GTNewHorizons/NotEnoughItems/assets/31038811/894f12b6-a139-4236-a32c-c4c7e55764aa)

![image](https://github.com/GTNewHorizons/NotEnoughItems/assets/31038811/f0592145-a67c-4dba-ab12-5ecfe36344bb)

Reply:
This is fine for now, but something we should look into in the future